app/parse.py

```diff
@@ -15,8 +39,81 @@ class Product:
     num_of_reviews: int
 
 
+def create_products(products_data: list[dict]) -> [Product]:
+    return [Product(**element) for element in products_data]
+
+
+def parse_element(element: WebElement) -> dict:
+    return {
+        "title": element.find_element(By.CLASS_NAME, "title").get_attribute("title"),
+        "description": element.find_element(By.CLASS_NAME, "description").text,
+        "price": float(
+            element.find_element(By.CLASS_NAME, "price").text.replace("$", "")
+        ),
+        "rating": len(element.find_elements(By.CLASS_NAME, "ws-icon-star")),
+        "num_of_reviews": element.find_element(
+            By.CLASS_NAME, "review-count"
+        ).text.split()[0],
+    }
+
+
+def parse_category(category_url: str) -> None:
+    driver.get(category_url)
+    while True:
+        try:
+            load_more_button = WebDriverWait(driver, 10).until(
+                expected_conditions.presence_of_element_located((By.CLASS_NAME, "ecomerce-items-scroll-more"))
+            )
+            if load_more_button.is_displayed() and load_more_button.is_enabled():
+                load_more_button.click()
+                time.sleep(2)
+            else:
+                break
+        except (NoSuchElementException, ElementNotInteractableException, TimeoutException):
+            print(f"Unable to find 'Load More' button or it's not interactable. Exiting...")
+            break
+
+    elements = driver.find_elements(By.CLASS_NAME, "thumbnail")
+    category_data = []
+    for element in elements:
+        category_data.append(parse_element(element))
+    return create_products(category_data)
+
+
+def parse_list_of_categories(list_categories: list[str]):
+    list_products = []
+
+    for category in list_categories:
+        list_products.append(parse_category(category))
+
+    return list_products
```

Comment:
The `parse_list_of_categories` function currently returns a list of lists, which may not be the intended behavior. Consider flattening the list before returning it, so that it returns a single list containing all `Product` objects from all categories.